### DISCUSSION TITLE
meshopt_simplify on uv seams

### DISCUSSION THREAD

**Discussion by craouette** [zeux/meshoptimizer#426] [Q&A] [answered]
Hi,

Is meshopt_simplify correctly handling uv seams?
suppose I have triangles like this:
```
1           2,7           3
+------------+------------+
|\           |\           |
|  \    A    |  \         |
|    \       |    \       |
|      \     |      \     |
|        \   |   B    \   |
|          \ |          \ |
+------------+------------+
4           5,8           6
```

edge 2->5 is an uv seams, so, in the index buffer, triangle A is 1, 2, 5 and triangle  B is 7, 6, 8, but the 3D coordinates of points 2 and 7 are identical (and the same for points 5 and 8)
The problem will be if point 2 is merged to point 5 but not point 7 to 8, then a hole will be visible.
Another point, is it possible to get, for one resulting triangle the list of source triangles? That would be useful to propagate face attributes to the resulting mesh.

thanks for help, and for the nice library.
Craouette

**Comment by zeux** [comment, marked as answer]
> suppose I have triangles like this:

What will happen in this case is that simplifier will automatically recognize the 2-5/7-8 edge as a seam, and 2,7 & 5,8 vertices as wedges of a single point. After this it will either *always* move both vertices along the seam, or move neither. This depends on the specifics of the topology - eg if your geometry is just two quads, then no movement will be performed, but if it's part of the larger mesh, simplifier can reduce the complexity of the UV seam by collapsing edges of that seam. No holes should appear.

As for triangle list, this is currently not part of the API. I have a TODO list that I've yet to go through that has this and some other extensions; it's probably not very difficult to hack this up in the meantime...
